### PR TITLE
VACMS-14205: Editors unable to save changes to Staff Profile pages 

### LINKED ISSUE #14205: Editors unable to save changes to Staff Profile pages
## Describe the defect
Helpdesk has received three separate reports of Staff Profiles not saving changes when 'published' . I have seen this myself on VA Sioux Falls and VA Jackson health care in Production and in Staging - In the Recent Changes list, my changes do not appear -- the third report comes from VA Greater Los Angeles health care.

## To Reproduce
1. Go to 'https://prod.cms.va.gov/va-jackson-health-care' and log in as log in as Camille.Ballentine@va.gov
2. Navigate to Staff Profiles ([Jason C. Cain i](https://staging.cms.va.gov/jackson-health-care/staff-profiles/jason-c-cain)s the prime example) 
3. Make a change, then Publish
4. See error - The change is not saved, and the edit does not appear in the 'Recent Changes' list.

## AC / Expected behavior
When a change is made to a Staff profile by an editor, that change should be saved upon Publish and/or Draft

## Screenshots
Camille has removed the 'Acting' from the title and the Phone number several times but the changes do not 'stick'
image

 ## Additional context
Other reports are from 

Amber.Bailey-Froke@va.gov at VA Sioux Falls health care
Lauren.Bolanos@va.gov at VA Greater Los Angeles health care

## Labels
(You can delete this section once it's complete)
- [x] Issue type (red) (defaults to "Defect")
- [ ] CMS subsystem (green)
- [ ] CMS practice area (blue)
- [x] CMS workstream (orange) (not needed for bug tickets)
- [ ] CMS-supported product (black)

### Team
Please check the team(s) that will do this work.

- [x] `CMS Team`
- [ ] `Public Websites`
- [ ] `Facilities`
- [ ] `User support`

### PR DESCRIPTION
**Pull request by edmund-dunn**
## Description

Closes #14205

## Testing done
Tested locally. All changes now save correctly. 

## QA steps

What needs to be checked to prove this works?  
What needs to be checked to prove it didn't break any related things?  
What variations of circumstances (users, actions, values) need to be checked?  

As user _uid_ with _user_role_
1. Go to 'https://prod.cms.va.gov/va-jackson-health-care' and log in as log in as Camille.Ballentine@va.gov
2. Navigate to Staff Profiles ([Jason C. Cain i](https://staging.cms.va.gov/jackson-health-care/staff-profiles/jason-c-cain)s the prime example) 
3. Make a change, then Publish
   - [x] Validate that changes save. 
2. Then navigate to https://va-gov-cms.ddev.site/jackson-health-care/staff-profiles/james-burrell and edit the node, delete the biography fields and make sure the Create profile page with bio checkbox is unchecked and save
   - [x] Validate that it saves correctly and that the Bio fields are in fact removed. 

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [x] `CMS Team`
- [ ] `Public websites`
- [ ] `Facilities`
- [ ] `User support`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing announcement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping the UX writer so they are ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping the UX writer to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

File: config/sync/field.field.node.person_profile.field_body.yml [hunk 1]
```diff
@@ -8,16 +8,19 @@ dependencies:
...
+  tmgmt_content:
+    excluded: false
 id: node.person_profile.field_body
 field_name: field_body
 entity_type: node
 bundle: person_profile
 label: 'Body text'
 description: 'Add everything else from the second sentence on. Don''t include the first sentence in this field.'
-required: true
+required: false
```
Comment (ndouglas) [inline config/sync/field.field.node.person_profile.field_body.yml:23]: Is this intended?

**Review by ndouglas — APPROVED** [review]
This seems to work for me. Just a question about the tmgmt changes.

**Comment by jilladams** [comment]
Hi @edmund-dunn @ndouglas until we get code reviewer groups stood up, we need help from y'all to tag in Product team Drupal folks for review, when PRs pertain to Facilities or PW products. An easy way to know is here: https://app.mural.co/t/departmentofveteransaffairs9999/m/vagov6717/1646745279937/9a5570c420939d84e875e8334fd73073e3a3971b?fromVisitorModal=true&sender=u0b235d03cbd64f7f93673243

In this case, Facilities needs to be looped in re: changes to staff profile. Adding @omahane and @swirtSJW as reviewers.

**Comment by jilladams** [comment]
Please hold on merging, til they've had a chance to weigh in.

File: config/sync/field.field.node.person_profile.field_body.yml (hunk 1 above)
Comment (jilladams) [reply config/sync/field.field.node.person_profile.field_body.yml:23]: a flag: this PR is approved, without an answer to this. @swirtSJW @omahane please advise re: whether this field should remain required.

Comment (ndouglas) [reply config/sync/field.field.node.person_profile.field_body.yml:23]: Sorry, that was my imprecise selection. I understand why the field is no longer required -- what I meant to ask about was the changes related to `tmgmt`.

File: config/sync/field.field.node.person_profile.field_body.yml [hunk 2]
```diff
@@ -8,16 +8,19 @@ dependencies:
   module:
     - allowed_formats
     - text
+    - tmgmt_content
```
Comment (ndouglas) [inline config/sync/field.field.node.person_profile.field_body.yml:11]: Is this intended?

File: config/sync/field.field.node.person_profile.field_body.yml [hunk 3]
```diff
@@ -8,16 +8,19 @@ dependencies:
...
 third_party_settings:
   allowed_formats:
     allowed_formats: {  }
+  tmgmt_content:
+    excluded: false
```
Comment (ndouglas) [inline config/sync/field.field.node.person_profile.field_body.yml:16]: Is this intended?

**Review by swirtSJW — COMMENTED** [review]
What was the root cause of this failure?  I looked that the issue and the slack thread and I don't see where it was mentioned what changed that suddenly broke this code that has been working fine for almost 2 years.

**Comment by ndouglas** [comment]
@swirtSJW The code never worked. See my comment [here](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/12187#issuecomment-1464125642).

File: config/sync/field.field.node.person_profile.field_body.yml (hunk 1 above)
Comment (edmund-dunn) [reply config/sync/field.field.node.person_profile.field_body.yml:23]: Yes.

File: config/sync/field.field.node.person_profile.field_body.yml (hunk 3 above)
Comment (edmund-dunn) [reply config/sync/field.field.node.person_profile.field_body.yml:16]: No, but came with it. I will remove.

**Review by swirtSJW — CHANGES_REQUESTED** [review]
This PR is essentially removing business logic without discussion or why or if it is appropriate to change the logic.

The business logic is that if a person checks off the box for making a FE Profile page, then it must have those two fields populated.
image


I understand that there is a bug that is getting hung up on the conditional requiredness of these fields, but the solution can not be to remove it without discussion.    We have other instances around the CMS of conditional requiredness.  Removing them all is not really an option.  I would like to understand what the underlying change was that caused this to start failing now after nearly 2 years of working correctly.

**Comment by swirtSJW** [comment]
> @swirtSJW The code never worked. See my comment [here](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/12187#issuecomment-1464125642).

@ndouglas  Maybe I am misunderstanding.  As far as I can tell it was working a month ago.

If I leave the box checked and try to save the node, I get a warning that those fields are required
image

And if I uncheck the box, it let me save it no problem.
image

Here is where I did that on a 1 month old demo environment https://main-1ouqufuiwhqohaqnvgjjoaidpjpensj4.demo.cms.va.gov/node/55457/revisions/view/765204/775480/visual_inline

**Pull request closed without merging**